app/controllers/admin/twitter_accounts_controller.rb

```diff
@@ -0,0 +1,74 @@
+class Admin::TwitterAccountsController < Admin::BaseController
+  before_filter :find_twitter_account, only: %i[show edit update destroy]
+
+  def index
+    @twitter_accounts = TwitterAccount.order(created_at: :desc)
+  end
+
+  def show; end
+
+  def new
+    @twitter_account = TwitterAccount.new
+  end
+
+  def create
+    @twitter_account = TwitterAccount.new(permitted_parameters)
+    @twitter_account.twitter_account_info =
+      TwitterClient.user(@twitter_account.screen_name).to_json
+
+    if @twitter_account.save
+      redirect_to admin_twitter_account_url(@twitter_account)
+    else
+      render action: :new
+    end
+  end
+
+  def edit; end
+
+  def update
+    if @twitter_account.update_attributes(permitted_parameters)
+      flash[:notice] = "Twitter account saved!"
+      redirect_to admin_twitter_account_url(@twitter_account)
+    else
+      render action: :edit
+    end
+  end
+
+  def destroy
+    if @twitter_account.present? && @twitter_account.destroy
+      flash[:info] = "Twitter account deleted."
+      redirect_to admin_twitter_accounts_url
+    else
+      flash[:error] = "Could not delete Twitter account."
+      redirect_to edit_admin_twitter_account_url(@twitter_account)
+    end
+  end
+
+  private
+
+  def permitted_parameters
+    params.require(:twitter_account).permit(
+      :active,
+      :address,
+      :append_block,
+      :city,
+      :consumer_key,
```

Comment:
Are we using multiple/different keys currently?

Reply:
Does this mean that when you sign up a new account, you have to go to a twitter developer panel to collect the API keys, etc? If so, that's fine, we just need to document it.

Reply:
> Are we using multiple/different keys currently?

We are — each account has its own set of key, token, and secrets

> Does this mean that when you sign up a new account, you have to go to a twitter developer panel to collect the API keys, etc?

Yup! Added validation to those fields and a note to the admin dashboard — 67699a84

<img width="838" alt="Screen Shot 2019-09-10 at 5 10 04 AM" src="https://user-images.githubusercontent.com/4433943/64600524-4732c280-d389-11e9-9fa2-8ef48cc8b5b9.png">

Reply:
Great. Thanks!!